backend-lambda/services/productService.js

```diff
@@ -19,25 +19,54 @@ const getAllProducts = async (pool, limit, offset) => {
 
 // NOTE: This function is untested against cockroach
 const updateProduct = async (pool, id, product) => {
-  let sets = [];
-  for (let key in product) {
-    sets.push(Format("%I = %L", key, product[key]));
-  }
+  poolCheck(pool);
+  productCheck(product);
+
+  const values = Object.values(product);
+  const keys = Object.keys(product);
 
-  let setStrings = sets.join(",");
+  // Creates keys and values array then maps them to one long-ass string ->
+  // ex: 'name = 'purple jumpsuit', size = 'Medium', description = 'comes with pockets', sold_date = '6.29.2024''
+  const setString = keys
+    .map((key, index) => Format("%I = %L", key, values[index]))
+    .join(", ");
 
-  const query = "UPDATE product SET %s WHERE product_id = %L";
-  const params = [setStrings, id];
+  /* 
+  The setString does not require a placeholder in the query nor to be in the parameters variable.
+  This is because the pg library formats the query so that the long-ass string is put into it before
+  it is passed to clientService ->
+  
+  ex:
+  UPDATE product SET name = 'purple jumpsuit', size = 'Medium', description = 'comes with pockets', sold_date = '6.29.2024'
+  WHERE product_id = $1
 
-  return clientService(pool, query, params);
+  The product_id value is the only placeholder used here with postgres's $1 placeholder syntax
+  */
+  const query = `UPDATE product SET ${setString} WHERE product_id = $1`;
+  const params = [id];
+
+  const res = await clientService(pool, query, params);
+  return res?.rows;
```

Comment:
Since you're not SELECTing anything, you're not really going to have any useful information here, if it's defined at all. I would check the log to see if there's anything useful there, otherwise feel free to just `return`

Reply:
nothing worth returning, changed to 
await client......;
return;